Server/sidebar.R

```diff
@@ -5,29 +5,27 @@
 # License: MIT License
 #####################################################################################################################
 
-# Start of the Observes'.
 
-# 1. Observe to update the radiobutton for decistion of the package.
-
-observe({
-  req(input$select_pack)
-  if (!identical(values$selected_pkg$decision, character(0))) {
-    if (values$selected_pkg$decision != "") {
-      updateSliderTextInput(
-        session,
-        "decision",
-        choices = c("Low", "Medium", "High"),
-        selected = values$selected_pkg$decision
-      )
-    }
+# Update the sidebar if a decision was previously made.
+observeEvent(input$select_pack, {
+  # Suppose package has been selected with a previously made decision.
+  if (nrow(values$selected_pkg) != 0 && values$selected_pkg$decision != "") {
+    # Update the risk slider using the info saved.
```

Comment:
Hi @marlycormar, if you want to change this to an observeEvent, you'll have to add  this event to the observer: `values$selected_pkg` plus a req() statement such as: `req(input$select_pack != "Select")`.

I tested it with only `input$select_pack` and `nrow(values$selected_pkg)` was producing an unexpected result: the values were always lagging behind whatever package I clicked in the UI. To confirm this, just add these print statements to the observeEvent:

`print(paste("values$selected_pkg:", values$selected_pkg))
  print(paste("nrow(values$selected_pkg):", nrow(values$selected_pkg)))`

I'll submit a small PR to this branch if this doesn't make sense. Thanks!

Reply:
Hi @marlycormar, I tried to push a new branch to your risk_assessment fork but got denied:

![image](https://user-images.githubusercontent.com/17878953/101361028-6f4ef100-386c-11eb-8e9c-0b667491089a.png)

I may have to be added as a collaborator if applicable. It's not a big issue though, the code is small enough to just paste it here:

![image](https://user-images.githubusercontent.com/17878953/101361175-a7563400-386c-11eb-9643-39ce3ba9dc01.png)

Hope that helps clarify.

Reply:
Good catch @AARON-CLARK!

I added lines 11 and 12 to the PR (line numbers from your screenshot). Do you think we still need like 18? The observeEvent should filter out empty inputs. I tested removing that conditional and it seems to work as expected.

Reply:
It seems to work without line 18!